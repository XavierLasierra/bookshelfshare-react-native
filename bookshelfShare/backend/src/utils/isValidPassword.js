const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
