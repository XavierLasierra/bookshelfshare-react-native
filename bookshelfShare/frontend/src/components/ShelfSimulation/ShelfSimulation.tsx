import React from 'react';
import { View } from 'react-native';
import ShelfBox from '../ShelfBox/ShelfBox';

import styles from './shelfSimulation.styles';

export default function ShelfSimulation({ shelfSize }: any) {
  function renderColumns(columns: number) {
    const renderedColumns = [];
    for (let i = 0; i < columns; i += 1) {
      renderedColumns.push(<ShelfBox width={`${100 / columns}%`} />);
    }
    return renderedColumns;
  }

  return (
    <>
      {shelfSize.map((columns: number) => (
        <View style={[styles.rowContainer, { height: `${100 / shelfSize.length}%`, width: '100%' }]}>
          {renderColumns(columns)}
        </View>
      ))}
    </>
  );
}
