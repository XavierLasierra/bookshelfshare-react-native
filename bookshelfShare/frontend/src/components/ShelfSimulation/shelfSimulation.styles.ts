import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  shelfFlexContainer: {
    height: 450,
    justifyContent: 'center'
  },
  rowContainer: {
    flexDirection: 'row'
  },
  shelfOutsideOutline: {
    borderWidth: 6,
    borderColor: stylesConstants.colors.shelfOutline,
    elevation: 10
  }
});
