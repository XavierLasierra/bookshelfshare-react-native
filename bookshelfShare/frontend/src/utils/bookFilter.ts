export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
