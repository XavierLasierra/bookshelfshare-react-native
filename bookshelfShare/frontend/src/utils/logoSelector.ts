import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
