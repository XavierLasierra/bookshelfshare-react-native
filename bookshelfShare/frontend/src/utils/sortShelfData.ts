export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
