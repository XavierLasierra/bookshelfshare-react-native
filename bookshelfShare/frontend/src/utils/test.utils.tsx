import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
