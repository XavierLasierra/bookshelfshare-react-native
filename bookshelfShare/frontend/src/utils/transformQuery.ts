interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
