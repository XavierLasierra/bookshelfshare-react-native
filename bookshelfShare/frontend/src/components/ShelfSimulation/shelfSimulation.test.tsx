import React from 'react';
import ShelfSimulation from './ShelfSimulation';
import { render } from '../../utils/test.utils';
import treatedShelfDataMock from '../../mocks/treatedShelfData.mock';

jest.useFakeTimers();

describe('Given a ShelfSimulation component', () => {
  describe('When it is rendered', () => {
    describe('And there are more rows or equal than columns', () => {
      test('Then should match the snapshot', () => {
        const screen = render(
          <ShelfSimulation
            clickCallback={jest.fn()}
            shelfSize={[2, 2]}
            shelfData={treatedShelfDataMock}
          />
        );
        expect(screen).toMatchSnapshot();
      });
    });
    describe('And there are less rows than columns', () => {
      test('Then should match the snapshot', () => {
        const screen = render(
          <ShelfSimulation
            clickCallback={jest.fn()}
            shelfSize={[1, 4]}
            shelfData={treatedShelfDataMock}
          />
        );
        expect(screen).toMatchSnapshot();
      });
    });
  });
});
