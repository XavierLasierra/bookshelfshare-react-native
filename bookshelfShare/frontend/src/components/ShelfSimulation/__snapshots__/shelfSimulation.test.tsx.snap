// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfSimulation component When it is rendered And there are less rows than columns Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "height": 450,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#958472",
        "borderColor": "#2F2406",
        "borderRadius": 15,
        "borderTopWidth": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#958472",
            "borderWidth": 6,
            "elevation": 10,
          },
          Object {
            "height": 200,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "height": "50%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-0-0"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              2
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "height": "50%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "25%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-0"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "25%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-1"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "25%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-2"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "25%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-3"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Given a ShelfSimulation component When it is rendered And there are more rows or equal than columns Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "height": 450,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#958472",
        "borderColor": "#2F2406",
        "borderRadius": 15,
        "borderTopWidth": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#958472",
            "borderWidth": 6,
            "elevation": 10,
          },
          Object {
            "height": 400,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "height": "50%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-0-0"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              2
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-0-1"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "height": "50%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-0"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#958472",
                "height": "100%",
                "justifyContent": "center",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4D3C2A",
                "borderBottomWidth": 5,
                "borderColor": "#2F2406",
                "borderRightWidth": 5,
                "height": "90%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
            testID="shelf-1-1"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              books
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
